lib/src/main/java/com/dynatrace/metric/util/Metric.java

```diff
@@ -23,6 +25,11 @@
 
   /** Builder class for {@link Metric Metrics}. */
   public static final class Builder {
+    private static final Logger logger = Logger.getLogger(Builder.class.getName());
+    // The timestamp warning is rate-limited to export only once every time this factor is reached
+    // by the timestampWarningCounter.
+    private static final short TIMESTAMP_WARNING_THROTTLE_FACTOR = 1000;
+    private static short timestampWarningCounter = 0;
```

Comment:
Since this state is shared even between multiple `MetricBuilderFactory`s, we should use an `AtomicInteger` to stay thread-safe. Even if it's just for the log throttle one would make wrong assumptions if it's erroneously logged twice or skipped.